forge-gui/res/cardsfolder/g/gilded_ambusher.txt

```diff
@@ -0,0 +1,14 @@
+Name:Gilded Ambusher
+ManaCost:1 B R
+Types:Enchantment Creature Lizard Glimmer
+PT:4/2
+T:Mode$ Attacks | ValidCard$ Card.Self | TriggerZones$ Battlefield | Execute$ TrigImmediateTrig | TriggerDescription$ Whenever CARDNAME attacks, you may sacrifice another nonland permanent. When you do, each opponent exiles a nonland, nontoken permanent they control and a random nonland card from their library, then CARDNAME deals damage to them equal to the total mana value of the cards exiled this way.
+SVar:TrigImmediateTrig:AB$ ImmediateTrigger | Cost$ Sac<1/Permanent.Other+nonLand/another nonland permanent> | Execute$ TrigEachOpp | TriggerDescription$ When you do, each opponent exiles a nonland, nontoken permanent they control and a random nonland card from their library, then CARDNAME deals damage to them equal to the total mana value of the cards exiled this way.
+SVar:TrigEachOpp:DB$ RepeatEach | RepeatPlayers$ Opponent | RepeatSubAbility$ DBChoose | SubAbility$ DBExile
+SVar:DBChoose:DB$ ChooseCard | Defined$ Remembered | Choices$ Permanent.nonLand+nonToken | ControlledByPlayer$ Chooser | ChoiceZone$ Battlefield | ImprintChosen$ True | SubAbility$ DBRandom
```

Comment:
doesn't it need Mandatory$?

Reply:
Probably! AI did always choose if it could but I didn't test to see if a human could cheat yet